packages/mco/components/dr-status-popover/parsers/subscription-parser.tsx

```diff
@@ -0,0 +1,228 @@
+import * as React from 'react';
+import { DRPCStatus, VolumeReplicationHealth } from '@odf/mco/constants';
+import {
+  DisasterRecoveryResourceKind,
+  getDRPlacementControlResourceObj,
+  getPlacementDecisionsResourceObj,
+  getPlacementResourceObj,
+  getPlacementRuleResourceObj,
+  getSubscriptionResourceObj,
+  SubscriptionGroupType,
+  useSubscriptionResourceWatch,
+} from '@odf/mco/hooks';
+import { useDisasterRecoveryResourceWatch } from '@odf/mco/hooks';
+import {
+  getDRClusterResourceObj,
+  getDRPolicyResourceObj,
+} from '@odf/mco/hooks';
+import {
+  findCluster,
+  getDRPolicyName,
+  getLastAppDeploymentClusterName,
+  getReplicationHealth,
+  getReplicationType,
+} from '@odf/mco/utils';
+import { getNamespace, ApplicationKind } from '@odf/shared';
+import DRStatusPopover from '../dr-status-popover';
+import { DRStatusProps } from './dr-parser-utils';
+
+const getMostSevereHealthStatus = (
+  drStatusList: {
+    healthStatus: VolumeReplicationHealth;
+  }[]
+): VolumeReplicationHealth => {
+  return drStatusList.reduce((acc: VolumeReplicationHealth, health) => {
+    if (health.healthStatus === VolumeReplicationHealth.CRITICAL) {
+      return VolumeReplicationHealth.CRITICAL;
+    } else if (
+      health.healthStatus === VolumeReplicationHealth.WARNING &&
+      acc !== VolumeReplicationHealth.CRITICAL
+    ) {
+      return VolumeReplicationHealth.WARNING;
+    }
+    return acc;
+  }, VolumeReplicationHealth.HEALTHY);
+};
+
+const parseDRStatusForGroup = (
+  group: SubscriptionGroupType
+): {
+  drPolicyName: string;
+  schedulingInterval?: string;
+  primaryCluster: string;
+  targetCluster?: string;
+  lastGroupSyncTime: string;
+  healthStatus: VolumeReplicationHealth;
+  phase: string;
+} | null => {
+  const drpc = group.drInfo?.drPlacementControl;
+  if (!drpc) return null;
+
+  const drPolicyName = getDRPolicyName(drpc);
+  const schedulingInterval = group.drInfo?.drPolicy?.spec?.schedulingInterval;
+
+  const primaryClusterName = getLastAppDeploymentClusterName(drpc);
+  const drClusters = group.drInfo?.drClusters;
+  if (!drClusters) {
+    return null;
+  }
+  const targetCluster = findCluster(drClusters, primaryClusterName);
+
+  const lastGroupSyncTime: string = drpc?.status?.lastGroupSyncTime;
+  const replicationType = getReplicationType(schedulingInterval);
+
+  const healthStatus = getReplicationHealth(
+    lastGroupSyncTime,
+    schedulingInterval,
+    replicationType
+  );
+
+  return {
+    drPolicyName,
+    schedulingInterval,
+    primaryCluster: primaryClusterName,
+    targetCluster: targetCluster?.metadata?.name,
+    lastGroupSyncTime,
+    healthStatus,
+    phase: drpc?.status?.phase,
+  };
+};
+
+const getDRResources = (
+  namespace: string
+): {
+  resources: {
+    drPolicies: ReturnType<typeof getDRPolicyResourceObj>;
+    drClusters: ReturnType<typeof getDRClusterResourceObj>;
+    drPlacementControls: ReturnType<typeof getDRPlacementControlResourceObj>;
+  };
```

Comment:
```suggestion
):  WatchResources['resources']
```
WatchResources from disaster-recovery.ts hooks